tests/wakunode2/test_validators.nim

```diff
@@ -187,23 +187,28 @@ suite "WakuNode2 - Validators":
           version: 2, timestamp: afterTimestamp, ephemeral: true)
         discard await nodes[i].publish(some(spamProtectedTopic), unsignedMessage)
 
-    # Wait for gossip
-    await sleepAsync(4.seconds)
-
     # Since we have a full mesh with 5 nodes and each one publishes 25+25+25+25+25 msgs
     # there are 625 messages being sent.
     # 125 are received ok in the handler (first hop)
     # 500 are are wrong so rejected (rejected not relayed)
-    check:
-      msgReceived == 125
-
+    
     var msgRejected = 0
-    for i in 0..<5:
-      for k, v in nodes[i].wakuRelay.peerStats.mpairs:
-        msgRejected += v.topicInfos[spamProtectedTopic].invalidMessageDeliveries.int
+    
+    # Active wait for the messages to be delivered across the mesh
+    for i in 0..<10:
+      msgRejected = 0
+      for i in 0..<5:
+        for k, v in nodes[i].wakuRelay.peerStats.mpairs:
+          msgRejected += v.topicInfos[spamProtectedTopic].invalidMessageDeliveries.int
+      
+      if msgReceived == 125 and msgRejected == 500:
+        break
+      else:
+        await sleepAsync(1.seconds)
```

Comment:
This is awesome! I suggest to change a little the wait to break sooner

```suggestion
    for i in 0..<100:
      msgRejected = 0
      for i in 0..<5:
        for k, v in nodes[i].wakuRelay.peerStats.mpairs:
          msgRejected += v.topicInfos[spamProtectedTopic].invalidMessageDeliveries.int
      
      if msgReceived == 125 and msgRejected == 500:
        break
      else:
        await sleepAsync(100.milliseconds)
```

Reply:
Done! Thank you!